### PR TITLE
Added Asserts and Messages to Asserts in std.algorithm.iteration

### PR DESCRIPTION
**Pull request by JackStouffer**
This is a continuation of https://github.com/dlang/phobos/pull/4568 and https://github.com/dlang/phobos/pull/4573

### CODE REVIEW

File: std/algorithm/iteration.d [hunk 1]
```diff
@@ -422,7 +422,10 @@ private struct _Cache(R, bool bidir)
             auto opSlice(size_t low, size_t high)
             in
             {
-                assert(low <= high);
+                assert(
+                    low <= high,
+                    "Attempting to slice a cache with a larger first argument than the second"
+                );
```
Comment (andralex) [inline std/algorithm/iteration.d]: Four lines is really overdoing it. Also the message is prolix.

```d
assert(low <= high, "Bounds error in cache slicing.");
```

Also you need to decide whether you always end with the period or not (you did so far).

**Comment by JackStouffer** [comment]
@andralex Fixed

**Comment by quickfur** [comment]
LGTM. Thanks!

**Comment by quickfur** [comment]
Auto-merge toggled on

**Pull request merged**